src/lavinmq/amqp/stream/stream_message_store.cr

```diff
@@ -86,8 +80,12 @@ module LavinMQ::AMQP
     private def offset_at(seg, pos, retried = false) : Tuple(Int64, UInt32, UInt32)
       return {@last_offset, seg, pos} if @size.zero?
       mfile = @segments[seg]
-      msg = BytesMessage.from_bytes(mfile.to_slice + pos)
-      offset = offset_from_headers(msg.properties.headers)
+      offset = @segment_first_offset[seg]
+      mfile.pos = 4
```

Comment:
This is not introduced, or only a problem here, but using the internal MFile pos wont be thread safe.